src/java.base/share/classes/javax/crypto/spec/HPKEParameterSpec.java

```diff
@@ -0,0 +1,457 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package javax.crypto.spec;
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import java.security.AsymmetricKey;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.Key;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.HexFormat;
+import java.util.Objects;
+
+/**
+ * This immutable class specifies the set of parameters used with a {@code Cipher} for the
+ * <a href="https://www.rfc-editor.org/info/rfc9180">Hybrid Public Key Encryption</a>
+ * (HPKE) algorithm. The <a href=
+ * "{@docRoot}/../specs/security/standard-names.html#cipher-algorithms">
+ * standard algorithm name</a> for the cipher is "HPKE".
+ * <p>
+ * In HPKE, the sender's {@code Cipher} is always initialized with the
+ * recipient's public key in {@linkplain Cipher#ENCRYPT_MODE encrypt mode},
+ * while the recipient's {@code Cipher} object is initialized with its own
+ * private key in {@linkplain Cipher#DECRYPT_MODE decrypt mode}.
+ * <p>
+ * An {@code HPKEParameterSpec} object can be provided at HPKE
+ * {@linkplain Cipher#init(int, Key, AlgorithmParameterSpec) cipher initialization}.
+ * <p>
+ * An {@code HPKEParameterSpec} object can be created in two ways:
+ * <ul>
+ * <li> {@link #of()} creates an instance with unspecified KEM, KDF, and AEAD
+ * algorithms, which will be determined by the implementation based on the key
+ * provided to {@code init()}. This instance can only be used by the sender.
+ * <li> {@link #of(int, int, int)} creates an instance with explicitly
+ * specified KEM, KDF, and AEAD algorithm identifiers. This instance can be
+ * used by both the sender and the receiver.
+ * </ul>
+ * The terms "KEM algorithm identifiers", "KDF algorithm identifiers", and
+ * "AEAD algorithm identifiers" refer to their respective numeric values
+ * (specifically, {@code kem_id}, {@code kdf_id}, and {@code aead_id}) as
+ * defined in <a href="https://www.rfc-editor.org/rfc/rfc9180.html#section-7">Section 7</a>
+ * of RFC 9180 and the
+ * <a href="https://www.iana.org/assignments/hpke/hpke.xhtml">IANA HPKE page</a>.
+ * This class has defined constants for the standard algorithm identifiers.
+ * For example, {@link #KEM_DHKEM_P_256_HKDF_SHA256}, {@link #KDF_HKDF_SHA256},
+ * and {@link #AEAD_AES_128_GCM}.
+ * <p>
+ * Once an {@code HPKEParameterSpec} object is created, additional methods
+ * are available to generate new {@code HPKEParameterSpec} objects with
+ * different features:
+ * <ul>
+ * <li>
+ * Application-supplied information can be provided using the
+ * {@link #info(byte[])} method by both sides.
+ * <li>
+ * If HPKE modes {@code mode_auth} or {@code mode_auth_psk} are used,
```

Comment:
This could be reworded as: "To use the HPKE modes {@code mode_auth} ..."

Reply:
OK.